full-service/src/service/transaction.rs

```diff
@@ -242,12 +251,20 @@ impl TransactionMemo {
     ) -> Result<Box<dyn MemoBuilder + Send + Sync>, WalletTransactionBuilderError> {
         match self {
             Self::Empty => Ok(Box::new(EmptyMemoBuilder::default())),
-            Self::RTH => {
+            Self::RTH(subaddress_index) => {
                 let mut memo_builder = RTHMemoBuilder::default();
-                memo_builder
-                    .set_sender_credential(SenderMemoCredential::from(&account_key.ok_or(
-                        WalletTransactionBuilderError::RTHUnavailableForViewOnlyAccounts,
-                    )?));
+                let account_key = account_key
+                    .ok_or(WalletTransactionBuilderError::RTHUnavailableForViewOnlyAccounts)?;
+                let sender_memo_credential = match subaddress_index {
+                    Some(subaddress_index) => {
+                        SenderMemoCredential::new_from_address_and_spend_private_key(
+                            &account_key.subaddress(*subaddress_index),
+                            account_key.subaddress_spend_private(*subaddress_index),
+                        )
+                    }
```

Comment:
Consider adding tests that exercise this path

Reply:
absolutely! thanks for taking a look at this :)

Reply:
Omg it actually worked O_O